frontend/catalyst/pennylane_extensions.py

```diff
@@ -428,7 +432,78 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+    scalar_out = True
+    return Grad(
+        _ensure_differentiable(f), grad_params=_check_grad_params(method, scalar_out, h, argnum)
+    )
+
+
+def jacobian(f: DifferentiableLike, *, method=None, h=None, argnum=None):
+    """A :func:`~.qjit` compatible Jacobian transformation for PennyLane/Catalyst.
+
+    This function allows the Jacobian of a hybrid quantum-classical function
+    to be computed within the compiled program.
+
+    .. warning::
+
+        Currently, higher-order differentiation or differentiation of non-QNode functions
+        is only supported by the finite-difference method.
+
+    .. note::
+
+        Any JAX-compatible optimization library, such as `JAXopt
+        <https://jaxopt.github.io/stable/index.html>`_, can be used
+        alongside ``jacobian`` for JIT-compatible variational workflows.
+        See the :doc:`/dev/quick_start` for examples.
+
+    Args:
+        f (Callable): a function or a function object to differentiate
+        method (str): The method used for differentiation, which can be any of
+    ``["fd", "defer"]``, where:
+
+            - ``"fd"`` represents first-order finite-differences for the entire hybrid
+              circuit,
+
+            - ``"defer"`` represents deferring the quantum differentiation to the method
+              specified by the QNode, while the classical computation is differentiated
+              using traditional auto-diff.
+
+        h (float): the step-size value for the finite-difference (``"fd"``) method
+        argnum (Tuple[int, List[int]]): the argument indices to differentiate
+
+    Returns:
+        Grad: A Grad object that denotes the Jacobian of a function.
```

Comment:
Interesting, I know this is copied from the original, but I think what we should emphasize here is that we return a callable that computes the Jacobian on the wrapped function for the given arguments. The fact that we return the Grad class, which itself is not part of the documentation, is not so relevant.

```suggestion
        Callable: A callable object that computes the Jacobian of the wrapped function for the given
                  arguments.
```

The same could be added to grad.

Reply:
💯